(maint) Look deeper for yum repo updates

mwaggett opened the pull request:
This commit changes the mindepth for the yum repo command to 3, instead of 2.
Previously, repo updates were happening at the top-level puppet5 directory,
which caused repo updates to take a super extra long time.

mwaggett committed:
(maint) Look deeper for yum repo updates

This commit changes the mindepth for the yum repo command to 3, instead of 2.
Previously, repo updates were happening at the top-level puppet5 directory,
which caused repo updates to take a super extra long time.

mwaggett commented:
Before (tried my best to point out the puppet5 repo; not sure why `uniq` didn't fix that..):
```
molly.waggett@weth:/opt/repository/yum/puppet5$ find "/opt/repository/yum" -mindepth 2 -name "*.rpm" | xargs -I {} dirname {} | uniq
/opt/repository/yum/el/6/i386
/opt/repository/yum/el/6/PC1/i386
/opt/repository/yum/el/6/PC1/SRPMS
/opt/repository/yum/el/6/PC1/x86_64
/opt/repository/yum/el/6/puppet5/i386
/opt/repository/yum/el/6/puppet5/x86_64
/opt/repository/yum/el/6/dependencies/i386
/opt/repository/yum/el/6/dependencies/SRPMS
/opt/repository/yum/el/6/dependencies/x86_64
/opt/repository/yum/el/6/devel/i386
/opt/repository/yum/el/6/devel/SRPMS
/opt/repository/yum/el/6/devel/x86_64
/opt/repository/yum/el/6/x86_64
/opt/repository/yum/el/6/products/i386
/opt/repository/yum/el/6/products/SRPMS
/opt/repository/yum/el/6/products/x86_64
/opt/repository/yum/el/7/PC1/i386
/opt/repository/yum/el/7/PC1/SRPMS
/opt/repository/yum/el/7/PC1/x86_64
/opt/repository/yum/el/7/PC1/aarch64
/opt/repository/yum/el/7/PC1/ppc64le
/opt/repository/yum/el/7/puppet5/i386
/opt/repository/yum/el/7/puppet5/x86_64
/opt/repository/yum/el/7/dependencies/SRPMS
/opt/repository/yum/el/7/dependencies/x86_64
/opt/repository/yum/el/7/devel/i386
/opt/repository/yum/el/7/devel/SRPMS
/opt/repository/yum/el/7/devel/x86_64
/opt/repository/yum/el/7/x86_64
/opt/repository/yum/el/7/products/i386
/opt/repository/yum/el/7/products/SRPMS
/opt/repository/yum/el/7/products/x86_64
/opt/repository/yum/el/5/PC1/i386
/opt/repository/yum/el/5/PC1/SRPMS
/opt/repository/yum/el/5/PC1/x86_64
/opt/repository/yum/el/5/dependencies/i386
/opt/repository/yum/el/5/dependencies/SRPMS
/opt/repository/yum/el/5/dependencies/x86_64
/opt/repository/yum/el/5/devel/i386
/opt/repository/yum/el/5/devel/SRPMS
/opt/repository/yum/el/5/devel/x86_64
/opt/repository/yum/el/5/products/i386
/opt/repository/yum/el/5/products/SRPMS
/opt/repository/yum/el/5/products/x86_64
/opt/repository/yum/puppet5 # <--
/opt/repository/yum/puppet5/el/6/i386
/opt/repository/yum/puppet5/el/6/SRPMS
/opt/repository/yum/puppet5/el/6/x86_64
/opt/repository/yum/puppet5/el/7/SRPMS
/opt/repository/yum/puppet5/el/7/x86_64
/opt/repository/yum/puppet5/el/7/aarch64
/opt/repository/yum/puppet5/el/7/ppc64le
/opt/repository/yum/puppet5/el/5/i386
/opt/repository/yum/puppet5/el/5/x86_64
/opt/repository/yum/puppet5 # <--
/opt/repository/yum/puppet5/sles/12/SRPMS
/opt/repository/yum/puppet5/sles/12/x86_64
/opt/repository/yum/puppet5/sles/12/ppc64le
/opt/repository/yum/puppet5/sles/11/i386
/opt/repository/yum/puppet5/sles/11/x86_64
/opt/repository/yum/puppet5 # <--
/opt/repository/yum/puppet5/fedora/26/x86_64
/opt/repository/yum/puppet5/fedora/24/i386
/opt/repository/yum/puppet5/fedora/24/x86_64
/opt/repository/yum/puppet5/fedora/25/i386
/opt/repository/yum/puppet5/fedora/25/x86_64
/opt/repository/yum/puppet5 # <--
/opt/repository/yum/puppet5/puppet5-nightly
/opt/repository/yum/puppet5/cisco-wrlinux/7/x86_64
/opt/repository/yum/puppet5/cisco-wrlinux/5/x86_64
/opt/repository/yum/puppet-nightly
/opt/repository/yum/sles/12/PC1/i386
/opt/repository/yum/sles/12/PC1/SRPMS
/opt/repository/yum/sles/12/PC1/x86_64
/opt/repository/yum/sles/12/PC1/ppc64le
/opt/repository/yum/sles/11/PC1/i386
/opt/repository/yum/sles/11/PC1/x86_64
/opt/repository/yum/fedora/f23/PC1/i386
/opt/repository/yum/fedora/f23/PC1/x86_64
/opt/repository/yum/fedora/f24/PC1/i386
/opt/repository/yum/fedora/f24/PC1/x86_64
/opt/repository/yum/fedora/f22/PC1/i386
/opt/repository/yum/fedora/f22/PC1/x86_64
/opt/repository/yum/fedora/f26/PC1/x86_64
/opt/repository/yum/fedora/f25/PC1/i386
/opt/repository/yum/fedora/f25/PC1/x86_64
/opt/repository/yum/fedora/f20/PC1/i386
/opt/repository/yum/fedora/f20/PC1/SRPMS
/opt/repository/yum/fedora/f20/PC1/x86_64
/opt/repository/yum/fedora/f20/dependencies/i386
/opt/repository/yum/fedora/f20/dependencies/SRPMS
/opt/repository/yum/fedora/f20/dependencies/x86_64
/opt/repository/yum/fedora/f20/devel/i386
/opt/repository/yum/fedora/f20/devel/SRPMS
/opt/repository/yum/fedora/f20/devel/x86_64
/opt/repository/yum/fedora/f20/products/i386
/opt/repository/yum/fedora/f20/products/SRPMS
/opt/repository/yum/fedora/f20/products/x86_64
/opt/repository/yum/fedora/f21/PC1/i386
/opt/repository/yum/fedora/f21/PC1/SRPMS
/opt/repository/yum/fedora/f21/PC1/x86_64
/opt/repository/yum/fedora/f21/products/i386
/opt/repository/yum/fedora/f21/products/SRPMS
/opt/repository/yum/fedora/f21/products/x86_64
/opt/repository/yum/cisco-wrlinux/7/PC1/i386
/opt/repository/yum/cisco-wrlinux/7/PC1/x86_64
/opt/repository/yum/cisco-wrlinux/5/PC1/i386
/opt/repository/yum/cisco-wrlinux/5/PC1/x86_64
```
After:
```
/opt/repository/yum/el/6/i386
/opt/repository/yum/el/6/PC1/i386
/opt/repository/yum/el/6/PC1/SRPMS
/opt/repository/yum/el/6/PC1/x86_64
/opt/repository/yum/el/6/puppet5/i386
/opt/repository/yum/el/6/puppet5/x86_64
/opt/repository/yum/el/6/dependencies/i386
/opt/repository/yum/el/6/dependencies/SRPMS
/opt/repository/yum/el/6/dependencies/x86_64
/opt/repository/yum/el/6/devel/i386
/opt/repository/yum/el/6/devel/SRPMS
/opt/repository/yum/el/6/devel/x86_64
/opt/repository/yum/el/6/x86_64
/opt/repository/yum/el/6/products/i386
/opt/repository/yum/el/6/products/SRPMS
/opt/repository/yum/el/6/products/x86_64
/opt/repository/yum/el/7/PC1/i386
/opt/repository/yum/el/7/PC1/SRPMS
/opt/repository/yum/el/7/PC1/x86_64
/opt/repository/yum/el/7/PC1/aarch64
/opt/repository/yum/el/7/PC1/ppc64le
/opt/repository/yum/el/7/puppet5/i386
/opt/repository/yum/el/7/puppet5/x86_64
/opt/repository/yum/el/7/dependencies/SRPMS
/opt/repository/yum/el/7/dependencies/x86_64
/opt/repository/yum/el/7/devel/i386
/opt/repository/yum/el/7/devel/SRPMS
/opt/repository/yum/el/7/devel/x86_64
/opt/repository/yum/el/7/x86_64
/opt/repository/yum/el/7/products/i386
/opt/repository/yum/el/7/products/SRPMS
/opt/repository/yum/el/7/products/x86_64
/opt/repository/yum/el/5/PC1/i386
/opt/repository/yum/el/5/PC1/SRPMS
/opt/repository/yum/el/5/PC1/x86_64
/opt/repository/yum/el/5/dependencies/i386
/opt/repository/yum/el/5/dependencies/SRPMS
/opt/repository/yum/el/5/dependencies/x86_64
/opt/repository/yum/el/5/devel/i386
/opt/repository/yum/el/5/devel/SRPMS
/opt/repository/yum/el/5/devel/x86_64
/opt/repository/yum/el/5/products/i386
/opt/repository/yum/el/5/products/SRPMS
/opt/repository/yum/el/5/products/x86_64
/opt/repository/yum/puppet5/el/6/i386
/opt/repository/yum/puppet5/el/6/SRPMS
/opt/repository/yum/puppet5/el/6/x86_64
/opt/repository/yum/puppet5/el/7/SRPMS
/opt/repository/yum/puppet5/el/7/x86_64
/opt/repository/yum/puppet5/el/7/aarch64
/opt/repository/yum/puppet5/el/7/ppc64le
/opt/repository/yum/puppet5/el/5/i386
/opt/repository/yum/puppet5/el/5/x86_64
/opt/repository/yum/puppet5/sles/12/SRPMS
/opt/repository/yum/puppet5/sles/12/x86_64
/opt/repository/yum/puppet5/sles/12/ppc64le
/opt/repository/yum/puppet5/sles/11/i386
/opt/repository/yum/puppet5/sles/11/x86_64
/opt/repository/yum/puppet5/fedora/26/x86_64
/opt/repository/yum/puppet5/fedora/24/i386
/opt/repository/yum/puppet5/fedora/24/x86_64
/opt/repository/yum/puppet5/fedora/25/i386
/opt/repository/yum/puppet5/fedora/25/x86_64
/opt/repository/yum/puppet5/puppet5-nightly
/opt/repository/yum/puppet5/cisco-wrlinux/7/x86_64
/opt/repository/yum/puppet5/cisco-wrlinux/5/x86_64
/opt/repository/yum/sles/12/PC1/i386
/opt/repository/yum/sles/12/PC1/SRPMS
/opt/repository/yum/sles/12/PC1/x86_64
/opt/repository/yum/sles/12/PC1/ppc64le
/opt/repository/yum/sles/11/PC1/i386
/opt/repository/yum/sles/11/PC1/x86_64
/opt/repository/yum/fedora/f23/PC1/i386
/opt/repository/yum/fedora/f23/PC1/x86_64
/opt/repository/yum/fedora/f24/PC1/i386
/opt/repository/yum/fedora/f24/PC1/x86_64
/opt/repository/yum/fedora/f22/PC1/i386
/opt/repository/yum/fedora/f22/PC1/x86_64
/opt/repository/yum/fedora/f26/PC1/x86_64
/opt/repository/yum/fedora/f25/PC1/i386
/opt/repository/yum/fedora/f25/PC1/x86_64
/opt/repository/yum/fedora/f20/PC1/i386
/opt/repository/yum/fedora/f20/PC1/SRPMS
/opt/repository/yum/fedora/f20/PC1/x86_64
/opt/repository/yum/fedora/f20/dependencies/i386
/opt/repository/yum/fedora/f20/dependencies/SRPMS
/opt/repository/yum/fedora/f20/dependencies/x86_64
/opt/repository/yum/fedora/f20/devel/i386
/opt/repository/yum/fedora/f20/devel/SRPMS
/opt/repository/yum/fedora/f20/devel/x86_64
/opt/repository/yum/fedora/f20/products/i386
/opt/repository/yum/fedora/f20/products/SRPMS
/opt/repository/yum/fedora/f20/products/x86_64
/opt/repository/yum/fedora/f21/PC1/i386
/opt/repository/yum/fedora/f21/PC1/SRPMS
/opt/repository/yum/fedora/f21/PC1/x86_64
/opt/repository/yum/fedora/f21/products/i386
/opt/repository/yum/fedora/f21/products/SRPMS
/opt/repository/yum/fedora/f21/products/x86_64
/opt/repository/yum/cisco-wrlinux/7/PC1/i386
/opt/repository/yum/cisco-wrlinux/7/PC1/x86_64
/opt/repository/yum/cisco-wrlinux/5/PC1/i386
/opt/repository/yum/cisco-wrlinux/5/PC1/x86_64
```

This does not affect nightlies (I checked, I promise).

e-gris commented:
uniq only works on sorted files, so:

```bash
for repodir in $(find "__REPO_PATH__" -mindepth 3 -name "*.rpm" | xargs -I {} dirname {} | sort | uniq) ; do
```

e-gris reviewed (changes requested):
Use sort before uniq or drop uniq

mwaggett committed:
(maint) Use `sort` before `uniq`

This commit adds `sort` to the yum repo command so that the `uniq` command
actually does something.

mwaggett commented:
@e-gris updated!

underscorgan commented:
@mwaggett that seems fine,
```
morgan@weth:/opt/repository/yum$ find . -mindepth 2 -maxdepth 2 -type f
./base/README.txt
./puppet-nightly/puppet-nightly-release-sles-11.noarch.rpm
./puppet-nightly/puppet-nightly-release-cisco-wrlinux-5.noarch.rpm
./puppet-nightly/puppet-nightly-release-el-6.noarch.rpm
./puppet-nightly/puppet-nightly-release-cisco-wrlinux-7.noarch.rpm
./puppet-nightly/puppet-nightly-release-el-5.noarch.rpm
./puppet-nightly/puppet-nightly-release-el-7.noarch.rpm
./puppet-nightly/puppet-nightly-release-fedora-26.noarch.rpm
./puppet-nightly/puppet-nightly-release-sles-12.noarch.rpm
```

Though, those should probably be symlinks not copies

mwaggett commented:
@underscorgan something barfed at me when I tried to do symlinks, so I just switched to copies.

underscorgan commented:
@mwaggett okay. I can help you through this more after I get back. There should be no reason these can't be symlinks
